assert/assertion_order.go

```diff
@@ -9,7 +9,7 @@ import (
 func isOrdered(t TestingT, object interface{}, allowedComparesResults []CompareType, failMessage string, msgAndArgs ...interface{}) bool {
 	objKind := reflect.TypeOf(object).Kind()
 	if objKind != reflect.Slice && objKind != reflect.Array {
-		return false
+		return Fail(t, fmt.Sprintf("object %T is not a collection", object), msgAndArgs...)
```

Comment:
Because maps are also collections:

```suggestion
		return Fail(t, fmt.Sprintf("object %T is not an ordered collection", object), msgAndArgs...)
```